### PR TITLE
:book: Add /agenda redirect in book

### PR DESCRIPTION
**Pull request by vincepri**
For this iteration, we'll keep using the 2023 agenda until the 2024 and the historical ones are available somewhere safe.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by vincepri** [comment]
Confirmed this works in preview `https://deploy-preview-9961--kubernetes-sigs-cluster-api.netlify.app/agenda`

**Comment by vincepri** [comment]
cc @CecileRobertMichon @richardcase we might want to collect providers' agendas this way as well?

File: docs/book/book.toml [hunk 1]
```diff
@@ -12,6 +12,8 @@ additional-css = ["theme/css/custom.css"]
 
 [output.html.redirect]
 "/tasks/upgrade.html" = "/tasks/upgrading-cluster-api-versions.html"
+"/agenda.html" = "/agenda/2023.html"
+"/agenda/2023.html" = "https://docs.google.com/document/d/1ushaVqAKYnZ2VN_aa3GyKlS4kEd6bSug13xaXOakAQI"
```
Comment (chrischdi) [inline docs/book/book.toml:16]: We will have to change the link / rehost the doc.
The doc is not readable anymore for folks [xref](https://kubernetes.slack.com/archives/C8TSNPY4T/p1704703777087059?thread_ts=1704306627.447759&cid=C8TSNPY4T).

**Pull request merged**